### DISCUSSION TITLE
Still respect to CMake variable CMAKE_BUILD_TYPE in scripts when use CMakePresets.json

### DISCUSSION THREAD

**Discussion by msclock** [microsoft/vscode-cmake-tools#3998] [Ideas]
Hi, thanks for this awesome cmake tools plugin. I always like it.

I noticed vscode launch button didn't work because there must exist CMAKE_BUILD_TYPE in CMakePresets.json even if CMAKE_BUILD_TYPE explicitly set in cmake scripts. please refer to repro code hash https://github.com/serious-scaffold/ss-cpp/commit/b800fa57d70556c54c83f700cabc89d29c33d5a1.

And I think respect to CMAKE_BUILD_TYPE in cmake scripts when use CMakePresets.json may be a workaround.

Related #3643, #2039

**Comment by gcampbell-msft** [comment]
Thanks for the feedback. I've moved #3643 to be in our "On Deck" milestone so that we hopefully can get to it quicker. However, I'll prefer to track that issue than this discussion post. Closing this post in favor of the issue. 

Thanks again!
